UI/EncyclopediaDetailPanel.cpp

```diff
@@ -2428,7 +2428,27 @@ namespace {
         // show image of planet environment at the top of the suitability report
         std::string planet_type = boost::lexical_cast<std::string>(planet->Type());
         boost::algorithm::to_lower(planet_type);
-        detailed_description += "<img src=\"encyclopedia/planet_environments/" + planet_type + ".png\"></img>";
+
+        namespace fs = boost::filesystem;
+        std::string filename;
+        int matches = 0;
+        fs::directory_iterator end_it;
+        for (fs::directory_iterator it(ClientUI::ArtDir() / "encyclopedia" / "planet_environments"); it != end_it; ++it) {
+            try {
+                if (fs::exists(*it) && !fs::is_directory(*it) && boost::algorithm::starts_with(it->path().filename().string(), planet_type)) {
+                    ++matches;
+                    if (planet_id % matches == 0) {
```

Comment:
Could you add some comments explaining what this code is doing? This part in particular I don't understand... you're incrementing the number of matches and then checking if the planet ID is divisible by that number... then doing so again for any additional files found? Seems like this could be simplified by looking up the files once per execution.

Is this intended to achieve a particular distribution of the possible images based on the number of factors each planet ID number has...?

Reply:
The idea is to pick a random file, by hashing the planet id, using reservoir sampling. That said, it would be simpler and clearer to just use a vector and then pick a random entry, so I'll change it to do that.